### PR TITLE
bib fixes for xml doc

### PR DESCRIPTION
**Pull request by danielskatz**

### CODE REVIEW

File: paper/paper.bib [hunk 1]
```diff
@@ -8,20 +8,17 @@
...
 	Year = {2019},
-	Bdsk-Url-1 = {github.com:artivis/manif.git},}
 
 @misc{Sophus,
 	Author = {Hauke Strasdat},
 	Title = {Sophus},
 	year = 2018,
-	howpublished = {[online] \url{https://github.com/strasdat/Sophus}},
+	url = {https://github.com/strasdat/Sophus},
 	Bdsk-Url-1 = {https://github.com/strasdat/Sophus},}
```
Comment (artivis) [inline paper/paper.bib]: This tag was removed for other entries. Is it problematic?

File: paper/paper.bib [hunk 2]
```diff
@@ -39,12 +36,9 @@ @inproceedings{wave_geometry
 
 @misc{eigenweb,
 	Author = {Ga\"{e}l Guennebaud and Beno\^{i}t Jacob and others},
-	Howpublished = {[online] \url{http://eigen.tuxfamily.org}},
 	Title = {Eigen v3},
-	Url = {http://eigen.tuxfamily.org/index.php?title=Main_Page},
-	Urldate = {2018-12-02},
+	Url = {http://eigen.tuxfamily.org},
 	Year = {2010},
```
Comment (artivis) [inline paper/paper.bib]:
```suggestion
	Year = {2010},}
```
`eigenweb` is now missing its closing parenthesis `}`.

File: paper/paper.bib [hunk 3]
```diff
@@ -59,49 +53,37 @@ @Inbook{Koranne2011
...
 	Author = {Chirikjian, G.S.},
-	Date-Added = {2019-01-31 12:40:38 +0100},
-	Date-Modified = {2019-01-31 12:41:30 +0100},
 	Isbn = {9780817649449},
 	Lccn = {2009933211},
 	Publisher = {Birkh{\"a}user Boston},
 	Series = {Applied and Numerical Harmonic Analysis},
 	Title = {Stochastic Models, Information Theory, and Lie Groups, Volume 2: Analytic Methods and Modern Applications},
 	Url = {https://books.google.ch/books?id=hZ1CAAAAQBAJ},
 	Year = {2011},
```
Comment (artivis) [inline paper/paper.bib]:
```suggestion
	Year = {2011},}
```

File: paper/paper.bib [hunk 4]
```diff
@@ -59,49 +53,37 @@ @Inbook{Koranne2011
...
 
 @book{BARFOOT-17-Estimation,
 	Author = {Barfoot, Timothy D.},
-	Date-Added = {2018-09-19 12:38:11 +0200},
-	Date-Modified = {2018-09-19 12:38:37 +0200},
 	Doi = {10.1017/9781316671528},
 	Place = {Cambridge},
 	Publisher = {Cambridge University Press},
 	Title = {State Estimation for Robotics},
 	Year = {2017},
```
Comment (artivis) [inline paper/paper.bib]:
```suggestion
	Year = {2017},}
```

File: paper/paper.bib [hunk 5]
```diff
@@ -8,20 +8,17 @@
...
 @misc{manif,
 	Author = {Jeremie Deray and Joan Sol\`a},
-	Date-Added = {2019-01-09 19:46:45 +0100},
-	Date-Modified = {2019-01-09 22:35:53 +0100},
-	Howpublished = {\url{https://github.com/artivis/manif}},
+	url = {https://github.com/artivis/manif},
 	Month = {January},
 	Read = {0},
 	Title = {\texttt{manif}: a small {C++} header-only library for {L}ie theory.},
 	Year = {2019},
```
Comment (artivis) [inline paper/paper.bib]:
```suggestion
	Year = {2019},}
```
`manif` is missing its closing parenthesis.

File: paper/paper.bib (hunk 1 above)
Comment (danielskatz) [reply paper/paper.bib]: Yes, howpublished is problematic - simply using url is not

File: paper/paper.bib (hunk 2 above)
Comment (danielskatz) [reply paper/paper.bib]: fixed

File: paper/paper.bib (hunk 3 above)
Comment (danielskatz) [reply paper/paper.bib]: fixed

File: paper/paper.bib (hunk 4 above)
Comment (danielskatz) [reply paper/paper.bib]: fixed

File: paper/paper.bib (hunk 5 above)
Comment (danielskatz) [reply paper/paper.bib]: fixed

**Comment by danielskatz** [comment]
Sorry for not checking these - I thought they would be simple, but ended up being more complex.  These are mostly issues in the xml that is submitted to crossref with the paper, and they are things that we should handle better.  One of them (howpublished vs url) is something we just changed in the example paper in our readthedocs site.

File: paper/paper.bib (hunk 1 above)
Comment (artivis) [reply paper/paper.bib]: I meant, is `Bdsk-Url-1` tag problematic? You have removed it in other bib entries but not here.

**Comment by artivis** [comment]
> Sorry for not checking these - I thought they would be simple, but ended up being more complex. These are mostly issues in the xml that is submitted to crossref with the paper, and they are things that we should handle better. One of them (howpublished vs url) is something we just changed in the example paper in our readthedocs site.

No problem at all, thanks for helping fixing this!

File: paper/paper.bib (hunk 1 above)
Comment (danielskatz) [reply paper/paper.bib]: sorry.  It's not really problematic, but anything extra right now seems to be confusing our system and getting into the XML data.  I'll remove it.

Comment (danielskatz) [reply paper/paper.bib]: ok, now removed

**Comment by artivis** [comment]
@joansola Do you mind having a quick look at this PR. Looks good to me. Also, don't mind the failing tests, they are fixed in #124.

File: paper/paper.bib [hunk 6]
```diff
@@ -59,49 +55,39 @@ @Inbook{Koranne2011
...
-	Eprint = {1812.01537},
-	Institution = {Institut de Rob{\`{o}}tica i Inform{\`{a}}tica Industrial},
-	Isbn = {1812.01537v4},
-	Number = {IRI-TR-18-01},
-	Title = {{A micro Lie theory for state estimation in robotics}},
-	Url = {https://arxiv.org/abs/1812.01537v4},
-	Year = {2018},
-	Bdsk-Url-1 = {https://arxiv.org/abs/1812.01537v4}}
+@article{Sola18,
+    author = {Sol{\`{a}}, Joan and Deray, Jeremie and Atchuthan, Dinesh},
```
Comment (joansola) [inline paper/paper.bib]: Names here are `Surname, Name`, while on other entreis are `Name Surname`

File: paper/paper.bib [hunk 7]
```diff
@@ -59,49 +55,39 @@ @Inbook{Koranne2011
...
 	Publisher = {Birkh{\"a}user Boston},
 	Series = {Applied and Numerical Harmonic Analysis},
 	Title = {Stochastic Models, Information Theory, and Lie Groups, Volume 2: Analytic Methods and Modern Applications},
 	Url = {https://books.google.ch/books?id=hZ1CAAAAQBAJ},
 	Year = {2011},
-	Bdsk-Url-1 = {https://books.google.ch/books?id=hZ1CAAAAQBAJ},}
+}
 
 @book{BARFOOT-17-Estimation,
 	Author = {Barfoot, Timothy D.},
```
Comment (joansola) [inline paper/paper.bib]: `Surname, Name` instead of `Name, Surname`

Comment (danielskatz) [reply paper/paper.bib]: the two allowed styles are ```surname, name``` or ```name surname```  there's no need to be consistent; both will be interpreted correctly

File: paper/paper.bib (hunk 6 above)
Comment (danielskatz) [reply paper/paper.bib]: see response below

File: paper/paper.bib (hunk 7 above)
Comment (joansola) [reply paper/paper.bib]: OK

**Comment by danielskatz** [comment]
Is this ok to merge now?  If so, can someone do that?  Then we can move on to the overall JOSS paper acceptance

**Comment by artivis** [comment]
@danielskatz yes, merging now. Thanks again :+1:

**Pull request merged**